plugin/interactive.go

```diff
@@ -56,27 +55,51 @@ func Prompt(msg string) (string, error) {
 	inFd := int(os.Stdin.Fd())
 	inGrp, err := tcGetpgrp(inFd)
 	if err != nil {
-		return "", fmt.Errorf("error getting process group controlling stdin: %v", err)
+		return fmt.Errorf("error getting process group controlling stdin: %v", err)
 	}
 	curGrp := unix.Getpgrp()
 
-	var v string
 	if inGrp == curGrp {
-		// We control stdin
+		return fn(ctx)
+	}
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT)
+	cancelCtx, cancel := context.WithCancel(ctx)
+	go func() {
+		<-sigCh
```

Comment:
This won't block forever, will it? Guessing the `Reset` below prevents that?

Reply:
Hmm, it might. I can probably do something about that.

Reply:
Should be addressed now.

Reply:
Why not just close `sigCh` in a defer? You'll still need to call the cancel func either way

Reply:
Oh, didn't think of it.